src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.ContentSearch/SearchResultsController.ts

```diff
@@ -33,21 +39,72 @@ module Orckestra.Composer {
                     TotalCount: Total,
                     SelectedSortBy,
                     AvailableSortBys,
+                    FacetsVisible: FacetsVisible,
                     Pagination: {
                         PagesCount: 1,
                         CurrentPage: 1,
                         PreviousPage: false,
                         NextPage: false,
                     },
+                    isOverriden: isOverriden,
                     isLoading: false
                 },
                 mounted() {
                     this.Pagination = this.getPagination(PagesCount)
                     self.eventHub.subscribe(ContentSearchEvents.SearchResultsLoaded, this.onSearchResultsLoaded);
+                    self.showFacetsService.getShowFacets().then(
+                        (value: boolean) => {
+                                FacetsVisible = value;
+                                if (!value) this.hideFacet(true);  // as an intial setup we hide the facet and ask for an update to be made  
+                           
+                        }, 
+                        (error: any) => {
+                            FacetsVisible = true;
+                            self.showFacetsService.setShowFacets(true);
+                        }
+                    );
+                    
                 },
                 computed: {
                 },
+                updated: function () {
+                    this.updateProductColumns();
+                },
                 methods: {
+                    hideFacet(update = false): void {
+                        document.getElementById("leftCol").classList.add("w-0-lg");
+                        document.getElementById("rightCol").classList.remove("col-lg-9");
+                        if(update) self.vueSearchResults.$data.FacetsVisible = FacetsVisible; // setting this will trigger the "updated" function above only if requested
+                    },
+                    showFacet(): void {
+                        document.getElementById("leftCol").classList.remove("w-0-lg");
+                        document.getElementById("rightCol").classList.add("col-lg-9");
+                    },
+                    toggleFacet(): void {
+                        if (FacetsVisible) {
+                            this.hideFacet();
+                        }
+                        else {
+                            this.showFacet();
+                        }
+                        FacetsVisible = !FacetsVisible;
+                        self.vueSearchResults.$data.FacetsVisible = FacetsVisible; // setting this will trigger the "updated" function above
```

Comment:
do you need both variables - global and inside components?
also inside component, you can assign values like this
this.FacetsVisible = !this.FacetsVisible  
instead of
self.vueSearchResults.$data.FacetsVisible = ...

Reply:
We need local and global variable because the local method "updated" will not be triggered if a local variable is not changed also it is used in SearchHelpers.cshtml

Going to try and removing the global one